_clients/go.md

```diff
@@ -11,21 +11,171 @@ The OpenSearch Go client lets you connect your Go application with the data in y
+        MaxRetries: 5,
+        RetryOnStatus: []int{502, 503, 504},
+    })
+```
+
+## Creating an index
+
+To create an OpenSearch index, use the `IndicesCreateRequest` method. You can construct a JSON object with custom settings as follows:
+
+```go
+settings := strings.NewReader(`{
+    'settings': {
+        'index': {
+            'number_of_shards': 1,
+            'number_of_replicas': 0
+            }
+        }
+    }`)
+
+res := opensearchapi.IndicesCreateRequest{
+    Index: "go-test-index1", 
+    Body:  settings,
+}
+```
+
+## Indexing a document
+
+You can index a document into OpenSearch as follows:
+
+```go
+document := strings.NewReader(`{
+    "title": "Moneyball",
+    "director": "Bennett Miller",
+    "year": "2011"
+}`)
+
+docId := "1"
+req := opensearchapi.IndexRequest{
+    Index:      "go-test-index1",
+    DocumentID: docId,
+    Body:       document,
+}
+insertResponse, err := req.Do(context.Background(), client)
+```
+
+## Performing bulk operations
+
+You can perform several operations at the same time by using the `Bulk` method of the client. The operations may be of the same type or of different types.
+
+```go
+blk, err := client.Bulk(
+		strings.NewReader(`
+    { "index" : { "_index" : "go-test-index1", "_id" : "2" } }
+    { "title" : "Interstellar", "director" : "Christopher Nolan", "year" : "2014"}
+    { "create" : { "_index" : "go-test-index1", "_id" : "3" } }
+    { "title" : "Star Trek Beyond", "director" : "Justin Lin", "year" : "2015"}
+    { "update" : {"_id" : "3", "_index" : "go-test-index1" } }
+    { "doc" : {"year" : "2016"} }
+`),
+	)
+```
+
+## Searching for documents
+
+The easiest way to search for documents is to construct a query string. The following code uses a `multi_match` query to search for "miller" in the title and director fields. It boosts the documents where "miller" is in the title field. 
+
+```go
+content := strings.NewReader(`{
+    "size": 5,
+    "query": {
+        "multi_match": {
+        "query": "miller",
+        "fields": ["title^2", "director"]
+        }
+    }
+}`)
+
+search := opensearchapi.SearchRequest{
+    Index: []string{"go-test-index1"},
+    Body: content,
+}
+
+searchResponse, err := search.Do(context.Background(), client)
+```
+
+## Deleting a document
+
+You can delete a document as follows:
+
+```go
+delete := opensearchapi.DeleteRequest{
+    Index:      "go-test-index1",
+    DocumentID: "1",
+}
+
+deleteResponse, err := delete.Do(context.Background(), client)
+```
+
+## Deleting an index
+
+You can delete an index as follows:
+
+```go
+deleteIndex := opensearchapi.IndicesDeleteRequest{
+    Index: []string{"go-test-index1"},
+}
+
+deleteIndexResponse, err := deleteIndex.Do(context.Background(), client)
+```
+
+## Sample program
+
+This sample program creates a client, adds an index with non-default settings, inserts a document, performs bulk operations, searches for the document, deletes the document, and finally deletes the index:
```

Comment:
Either delete "finally" or place commas around it ("...and, finally, deletes...").